internal/plugin.go

```diff
@@ -136,6 +139,10 @@ func (p *Plugin) Generate() ([]byte, error) {
 				return nil, err
 			}
 
+			if len(validation.IsDNS1123Subdomain(plcName)) > 0 {
```

Comment:
In my original review, I didn't realize these checks weren't happening in `assertValidConfig`. Could you please change this?